imblearn/utils/_validation.py

```diff
@@ -27,6 +27,42 @@
 TARGET_KIND = ("binary", "multiclass", "multilabel-indicator")
 
 
+class _OutputReconstructor:
```

Comment:
This look nice. We could only add some tests in validation just for this class.

Reply:
Well, we could named it `OutputFormatter`. I think is a better name. We could also avoid the leading underscore. It is located in a internal module so it is not intended for public use. what do you think?

Reply:
@glemaitre I added tests that they check the intention of the formatter

Reply:
Actually it does not format. It converts/transforms numpy arrays to their original types. So,  `OutputConverter`, `OutputTransformer`, `DataConverter`, `DataTransfromer`, `ArraysConverter` or `ArraysTransformer` would be a better choice, what do you think?

Reply:
Arraysttansformer would be best for me

Reply:
Ok. Let's named it like this.